Make use of apostrophe and \prime consistent.

elsanussi-s-mneina opened the pull request:
I am not sure of whether the changes I made in lines 54 to 58 of Colist.agda will affect code outside the library. This is because it uses the GHC directive. Please double check the changes I made to that file.

elsanussi-s-mneina commented:
This pull request is an attempt to alleviate issue #995.

elsanussi-s-mneina commented:
OK, it failed due to this error: Failed to parse GHC pragma '= data AgdaColist′ (Nil | Cons)'
3450. I shouldn't have changed that apostrophe to a prime. What I don't understand is how I didn't notice the tests fail. Maybe the setup on my machine is wrong. I did run, make clean then make test. I will try again, and try again next week.

gallais commented:
> Maybe the setup on my machine is wrong. I did run, make clean then make test.

Travis is running more tests, including basic compilations ones (although we
would like to get more comprehensive testing of compiling the stdlib, cf. #289)

MatthewDaggitt commented:
Thanks for the PR! Github helpfully uses the same symbol so I can't tell which way you went? I would almost certainly vote in favour of apostrophe rather than prime because it's so much easier to type.

There's also the annoying issue that we should almost certainly deprecate the old names rather than simply eliminating them (see e.g. `Data.Nat.Properties`). In this instance the deprecation message should also make clear what's changed (i.e. using ' rather than \prime) as it won't be obvious to everyone!

elsanussi-s-mneina commented:
Hi, I went in favour of the prime. So the deprecation message would be (using \prime rather than ' (apostrophe)). I thought it would be more mathematically accurate since it is called prime, but I know it makes it a pain if the font is not large enough. Of course, Haskell does not allow use of prime in identifiers, and some of the code in agda-std-lib references Haskell identifiers that have an apostrophe, I could not change those. My mistake was changing those references at first.

Should I attempt to deprecate the old names in this pull request? 
Aside, I think that the MAlonzo folder should be in the .gitignore file. It seems like it is generated as part of the build.

I modified the Changelog, but I am unsure whether the message went in the best place. I simply put it at the end.

I still need to do the deprecation. Perhaps tomorrow.

MatthewDaggitt commented:
> I thought it would be more mathematically accurate since it is called prime, but I know it makes it a pain if the font is not large enough. 

I guess that's true. And it looks like the top-level name changes are much smaller this way round so it'll have less effect on users. Consider me convinced :+1: 

> Should I attempt to deprecate the old names in this pull request?

Yes please. If you could add a [WARNING_ON_USAGE](https://github.com/agda/agda-stdlib/blob/9f454e2320a02249aa0c03f08e40217be1ab7e9d/src/Data/Nat/Properties.agda#L2260) pragma to each top-level name changed that would be great. It just gives users a much better experience than us breaking their code! 

> Aside, I think that the MAlonzo folder should be in the .gitignore file. It seems like it is generated as part of the build.

Agreed. It's minor enough that I'm happy for it to be added to this PR. Alternatively you can open another if you'd like.

MatthewDaggitt commented on CHANGELOG.md:
```diff
@@ -8,6 +8,10 @@ Highlights
 
+* New standardised numeric predicates `NonZero`, `Positive`, `Negative`,
+  `NonPositive`, `NonNegative`, especially designed to work as instance
+  arguments.
```
Something slightly strange seems to have happened to the CHANGELOG. These additions shouldn't be here. A merge mistake somewhere?

elsanussi-s-mneina replied:
There was a conflict in CHANGELOG.md, so I came up with a plan that did not work. I went into the master branch, and copied the latest version of the CHANGELOG.md, and then pasted those into my branch, and then modified my changes to fit in there.    I wasn't sure if I was supposed to resolve the merges myself, but I guess not since it did not work. So the strangeness is probably due to my strange attempt.

elsanussi-s-mneina replied:
I will revert the CHANGELOG, and try again.

elsanussi-s-mneina replied:
I reverted the CHANGELOG.md file to the state it was at when I forked the repository, and it seems to have resolved the conflict. Of course, I will have to make the changes again or else there will be no information on the changes I made in this pull request.   

Another guess I have for the cause of the conflict is that I may have changed the line   

* Added new proofs to ``Data.Sum.Properties`:

To have only one backtick between "to" and "Data". It seems erroneous to have two backticks there according to my understanding of Markdown syntax.

elsanussi-s-mneina commented:
> Yes please. If you could add a [WARNING_ON_USAGE](https://github.com/agda/agda-stdlib/blob/9f454e2320a02249aa0c03f08e40217be1ab7e9d/src/Data/Nat/Properties.agda#L2260) pragma to each top-level name changed that would be great. It just gives users a much better experience than us breaking their code!
> 
> >

I added the warning on usage. 

The part I am unsure of is whether changing the name of module parameters in Agda is a breaking change. I am not very familiar with module parameters. I changed the name of module parameters in the following files in case someone wants to take a look.
-  agda-stdlib/src/Function/Nary/NonDependent.agda (line 71)
- /Users/elsanussi/Downloads/Code/agda-stdlib/src/Data/Vec/Properties.agda (line 300)
/Users/elsanussi/Downloads/Code/agda-stdlib/src/Data/Tree/AVL/IndexedMap.agda (line 48)
/Users/elsanussi/Downloads/Code/agda-stdlib/src/Data/Vec/Properties.agda (line 300)

I think the adding of the MAlonzo file to the .gitignore should be done in a separate pull request, because there was also another file that I thought should be added (named "Haskell"), but am not sure of.

MatthewDaggitt commented on src/Data/List/Base.agda:
```diff
@@ -357,17 +357,21 @@ xs ∷ʳ? x = maybe′ (xs ∷ʳ_) xs x
+
+infixl 5 _∷ʳ'_
+_∷ʳ'_ : (xs : List A) (x : A) → InitLast (xs ∷ʳ x)
+_∷ʳ'_ = InitLast._∷ʳ′_
```
We usually put deprecated terms at the end of the file, just above the `WARNING_ON_USAGE ` pragma. We also don't usually include their type declaration (though it's important to keep their infix declaration.  The same goes for the other deprecated terms.

elsanussi-s-mneina replied:
I noticed that, but I could not get it to compile if I put the deprecated item at the end or omitted its type declaration. I am not sure why though. I can try again.

elsanussi-s-mneina replied:
In response, I put them at the end of the file, just above the WARNING_ON_USAGE pragma. I will try to remove the type declaration if the build succeeds.

gallais commented on README/Nary.agda:
```diff
@@ -147,8 +147,8 @@ curry₁ = curryₙ 4
-curry₁' : (A × (B × C) × D → E) → A → (B × C) → D → E
-curry₁' = curryₙ 3
+curry₁′ : (A × (B × C) × D → E) → A → (B × C) → D → E
+curry₁′ = curryₙ 3
```
It's in a README so I think it's fine.

MatthewDaggitt replied:
Whoops, yes! Please ignore :)

elsanussi-s-mneina commented on CHANGELOG/v0.15.md:
```diff
@@ -24,7 +24,7 @@ Non-backwards compatible changes
   `Relation.Binary.Sum`                  ↦ `Data.Sum.Relation.Pointwise`
                                          ↘ `Data.Sum.Relation.LeftOrder`
-  `Relation.Binary.Sigma.Pointwise`      ↦ `Data.Product.Relation.Pointwise.Dependent'
+  `Relation.Binary.Sigma.Pointwise`      ↦ `Data.Product.Relation.Pointwise.Dependent`
```
I have undone that change.

elsanussi-s-mneina commented on CHANGELOG/v0.16.md:
```diff
@@ -551,9 +551,9 @@ Other minor additions
-  ++-injective      : (xs xs' : Vec A n) → xs ++ ys ≡ xs' ++ ys' → xs ≡ xs' × ys ≡ ys'
+  ++-injectiveˡ     : (xs xs′ : Vec A n) → xs ++ ys ≡ xs′ ++ ys′ → xs ≡ xs′
+  ++-injectiveʳ     : (xs xs′ : Vec A n) → xs ++ ys ≡ xs′ ++ ys′ → ys ≡ ys′
+  ++-injective      : (xs xs′ : Vec A n) → xs ++ ys ≡ xs′ ++ ys′ → xs ≡ xs′ × ys ≡ ys′
```
I have undone that change.

elsanussi-s-mneina commented:
I have added MAlonzo to .gitignore as discussed.

elsanussi-s-mneina commented:
I just noticed if you go to line 2262 of [Data.Nat.Properties Line 2262](https://github.com/agda/agda-stdlib/blob/9f454e2320a02249aa0c03f08e40217be1ab7e9d/src/Data/Nat/Properties.agda#L2262), the pragma is wrong the name of what is deprecated does not match the deprecated item right above. It looks like a copy paste error.

elsanussi-s-mneina commented:
What do you suggest for dealing with the Conflicting files. I think they are all in the README folder. Should I merge the changes made from the commit in the master branch that updated the files in README?

elsanussi-s-mneina commented:
I merge the changes made from the commit in the master branch that updated the files in README.

MatthewDaggitt commented:
> I just noticed if you go to line 2262 of Data.Nat.Properties Line 2262, the pragma is wrong the name of what is deprecated does not match the deprecated item right above. It looks like a copy paste error.

That is indeed a copy and paste error! I'll open an issue to fix it.

Hmm your merge seems to have dragged in some extra changes from commit 0f437ab. Otherwise it looks good to go and after that's fixed I'll merge it in. Thanks again for the PR!

elsanussi-s-mneina commented:
I dragged in changes from that commit because in order to resolve a merge conflict. I'm not sure if that is the correct protocol, I'm new to this collaboration via git. I had to roll back some of the commits I made after merging 0f437ab, those commits were for fixing one apostrophe, and one incorrect character in the comment.

MatthewDaggitt commented:
Err so I think the best course of action is to rebase your branch on top of the latest changes from the library. If your [remote](https://help.github.com/en/github/using-git/adding-a-remote) for the standard library repo is called `upstream` then I would run:
```
git fetch upstream
git rebase origin/master
git push -f
```
which should get rid of the extra commits that are already present on the this repository's `master` branch.

elsanussi-s-mneina commented:
Sorry, for the question which might seem obvious. But I need to make sure I don't misunderstand. Are those three commands something I should have done? Or are they something I should do now also?

MatthewDaggitt commented:
> Sorry, for the question which might seem obvious. But I need to make sure I don't misunderstand. Are those three commands something I should have done? Or are they something I should do now also?

No worries, asking is always good :+1: So I think you _should have_ run the commands I listed above except with `merge` rather than `rebase`. I think (famous last words) that if you run them now with `rebase` then it should fix the problem.

elsanussi-s-mneina commented:
I'm going to start over, but first I will deal with some easier issues. I will be deleting my fork so I can start on a new one. It will be easier if I deal with the deprecation warnings, and the gitignore file in a separate pull request. Then I can resolve the apostrophe and \prime consistent without having to deal with too many changes.
